inttest/windows/windows_test.go

```diff
@@ -0,0 +1,241 @@
+// SPDX-FileCopyrightText: 2025 k0s authors
+// SPDX-License-Identifier: Apache-2.0
+
+package windows
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type WindowsSuite struct {
+	suite.Suite
+	kc         *kubernetes.Clientset
+	restConfig *rest.Config
+}
+
+func TestWindowsSuite(t *testing.T) {
+	suite.Run(t, new(WindowsSuite))
```

Comment:
I wonder if this would be better written without using `testify/suite`. Having t.Cleanup() and t.Context() in place allows for better test resource management than what we get from suites.